File: snippets/fsharp/System/UInt64/MaxValue/MaxValue1.fs
// <Snippet1>
open System

let decimalValue = -1.5

// Discard fractional portion of Double value
let decimalInteger = floor decimalValue

if decimalInteger <= float UInt64.MaxValue && decimalInteger >= float UInt64.MinValue then
    let integerValue = uint64 decimalValue
    printfn $"Converted {decimalValue} to {integerValue}."
else
    let rangeLimit, relationship =
        if decimalInteger > float UInt64.MaxValue then
            UInt64.MaxValue, "greater"
        else
            UInt64.MinValue, "less"

    printfn $"Conversion failure: {decimalInteger} is {relationship} than {rangeLimit}."
// </Snippet1>

File: snippets/csharp/System/UInt64/MaxValue/MaxValue1.cs
using System;

public class ULongRangeExample
{
    public static void Main()
    {
        // <Snippet1>
        double decimalValue = -1.5;
        ulong integerValue;

        // Discard fractional portion of Double value
        double decimalInteger = Math.Floor(decimalValue);

        if (decimalInteger <= ulong.MaxValue &&
            decimalInteger >= ulong.MinValue)
        {
            integerValue = (ulong)decimalValue;
            Console.WriteLine($"Converted {decimalValue} to {integerValue}.");
        }
        else
        {
            ulong rangeLimit;
            string relationship;

            if (decimalInteger > ulong.MaxValue)
            {
                rangeLimit = ulong.MaxValue;
                relationship = "greater";
            }
            else
            {
                rangeLimit = ulong.MinValue;
                relationship = "less";
            }

            Console.WriteLine($"Conversion failure: {decimalInteger} is {relationship} than {rangeLimit}.");
        }
        // </Snippet1>
    }
}
